### PR TITLE
Add delete button to new order item

### LINKED ISSUE #15: [Feature request]: Add `Delete` button on new order lines
### Describe the proposed solution

image

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Zales0123**
Fixes #15 

https://github.com/Setono/sylius-order-edit-plugin/assets/6212718/2f35e155-ef1e-4470-8a9c-cb6ae34f8e38

What still don't work are discounts for the new order item - but it was not in a issue description :dancer: So didn't want to be too hasty with implementing such a feature (and, obviously, the delete button is more crucial)

### CODE REVIEW

File: src/Resources/public/js/order-edit.js [hunk 1]
```diff
@@ -1,24 +1,32 @@
-window.addEventListener('DOMContentLoaded', (event) => {
-    document.querySelectorAll('button.delete-order-item').forEach((button) => {
-        button.addEventListener('click', (event) => {
-            const orderLineId = event.currentTarget.closest('tr').remove();
-        });
+document.querySelectorAll('button.delete-order-item').forEach((button) => {
+    button.addEventListener('click', (event) => {
+        var row = event.currentTarget.parentElement.parentElement;
```
Comment (loevgaard) [inline src/Resources/public/js/order-edit.js]: Use `event.currentTarget.closest('..')`

**Review by loevgaard — APPROVED** [review]

File: src/Resources/public/js/order-edit.js (hunk 1 above)
Comment (Zales0123) [reply src/Resources/public/js/order-edit.js]: Zrzut ekranu 2024-06-27 o 11 24 57

hmm, doesn't seems to work :/

Comment (loevgaard) [reply src/Resources/public/js/order-edit.js]: I don't understand how that's possible :D

File: src/Setter/OrderDiscountAdjustmentSetter.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        private readonly IntegerDistributorInterface $integerDistributor,
+        private readonly DiscountAdjustmentsAdderInterface $orderItemDiscountAdjustmentAdder,
+    ) {
+    }
+
+    public function set(OrderInterface $order, int $discount): void
+    {
+        $items = $order->getItems();
+        /** @var int $orderId */
+        $orderId = $order->getId();
```
Comment (loevgaard) [inline src/Setter/OrderDiscountAdjustmentSetter.php]:
```suggestion
        $orderId = (int) $order->getId();
```

**Review by loevgaard — APPROVED** [review]

**Pull request merged**